composer/trainer/trainer.py

```diff
@@ -2244,6 +2247,14 @@ def _train_loop(self) -> None:
 
                 self.engine.run_event(Event.EPOCH_CHECKPOINT)
 
+                # Increment iteration
+                if (self.state._iteration_length is not None and
+                        self.state.timestamp.epoch_in_iteration == self.state._iteration_length):
+                    self.state.previous_timestamp = self.state.timestamp
+                    self.state.timestamp = self.state.timestamp.to_next_iteration()
```

Comment:
Where is `to_next_iteration()` implemented? If it a part of this PR or am I missing sth?

Reply:
It got added in #3017